colossus/src/main/scala/colossus/service/ServiceClient.scala

```diff
@@ -273,6 +275,8 @@ extends Controller[O,I](codec, ControllerConfig(config.pendingBufferSize, config
       //don't allow any new requests, appear as if we're dead
       s.handler(Failure(new NotConnectedException("Not Connected")))
     } else if (isConnected || !failFast) {
+      requestBufferLoad.set(queueSize)
```

Comment:
So..I'm a bit confused.  These metrics are for the pending buffer(not yet sent).  However the only time they are being ticked is in purgeBuffers, which is called only on connectionLost/Closed(from what i'm gathering here).  Is this intentional?  Do you only care about the pending queue size when the connection is lost or closed?

If so, they should be renamed to something along those lines.

If not, then we need to also think through when to tick these metrics(especially the gauge), when we enqueue and dequeue pending requests.